src/analysis/transcendental/deriv.lean

```diff
@@ -0,0 +1,247 @@
+        {
+            apply congr_arg, rw function.funext_iff, intro i, ring,
+        },
+
+        transitivity
+            (∑ (x : ℕ) in finset.range n,
+                ((polynomial.C (n.choose x + (n.choose (x + 1)):ℤ)) * (deriv_n p (n - x)) * deriv_n q (x + 1))),
+        {
+            apply congr_arg, rw function.funext_iff, intro i,
+            simp only [int.cast_add, ring_hom.eq_int_cast],
+        },
+
+        transitivity
+            (∑ (x : ℕ) in finset.range n,
+                ((polynomial.C ((n+1).choose (x + 1):ℤ)) * (deriv_n p (n - x)) * deriv_n q (x + 1))),
+        {
+            apply congr_arg, rw function.funext_iff, intro i,
+            rw [nat.choose_succ_succ, int.coe_nat_add],
+        },
+
+        conv_rhs {rw finset.sum_range_succ', rw finset.sum_range_succ},
+        simp only [deriv_succ, zeroth_deriv, nat.succ_eq_add_one, nat.choose_self, int.cast_coe_nat,
+            ring_hom.eq_int_cast, one_mul, nat.succ_sub_succ_eq_sub, nat.choose_zero_right,
+            int.coe_nat_zero, nat.sub_self, int.cast_one, int.coe_nat_succ, nat.sub_zero, zero_add],
+        ring,
+    }
+end
+
+/-Theorem
+For a polynomial $f$ then if $n>0$, we have $f^{(n)}=f^{(n-1)}\times f'$
+-/
+
+theorem poly_pow_deriv (f : ℤ[X]) (n : ℕ) (hn : 0 < n) : (f ^ n).derivative = (polynomial.C (n:ℤ)) * (f ^ (n-1)) * f.derivative :=
+begin
+    induction n with n IH,
+    exfalso, linarith,
+    {
+        cases n,
+        simp only [ring_hom.eq_int_cast, one_mul, int.coe_nat_zero, int.cast_one, int.coe_nat_succ, pow_one, zero_add, pow_zero],
+        replace IH := IH (nat.succ_pos n),
+        rw nat.succ_eq_add_one, rw pow_add, simp only [int.cast_coe_nat, int.cast_add, ring_hom.eq_int_cast, polynomial.derivative_mul, int.cast_one, nat.succ_add_sub_one,
+ int.coe_nat_succ, pow_one], rw IH, simp only [nat.succ_sub_succ_eq_sub, polynomial.C_add, polynomial.C_1, int.coe_nat_succ, nat.sub_zero, nat.succ_sub_succ_eq_sub,
+ int.coe_nat_succ, nat.sub_zero],
+        have eq1 : (polynomial.C ↑n + 1) * f ^ n * f.derivative * f = (polynomial.C ↑n + 1) * f ^ (n+1) * f.derivative,
+        {
+            rw pow_add, simp only [int.cast_coe_nat, ring_hom.eq_int_cast, pow_one], ring,
+        } ,
+        rw eq1, rw nat.succ_eq_add_one, repeat {rw add_mul}, simp only [int.cast_coe_nat, ring_hom.eq_int_cast, one_mul],
+    }
+end
+
+theorem deriv_n_C_mul (c : ℤ) (n : ℕ) (f : ℤ[X]) :
+  (deriv_n (polynomial.C c * f) n) = (polynomial.C c) * (deriv_n f n) :=
+begin
+  rw [deriv_n, deriv_n, polynomial.iterate_derivative_C_mul],
+end
+
+theorem dvd_poly_pow_deriv (f : ℤ[X]) (n m : ℕ) (c : ℤ) (hn : 0 < n) (hm : 0 < m) :
+  (n:ℤ) ∣ polynomial.eval c (deriv_n (f^n) m) :=
+begin
+    cases m, linarith,
+
+    rw [deriv_n, function.iterate_succ], simp only [function.comp_app], rw [<-deriv_n], rw poly_pow_deriv,
+    have triv : polynomial.C ↑n * f ^ (n - 1) * f.derivative = polynomial.C ↑n * (f ^ (n - 1) * f.derivative) := by ring,
```

Comment:
`triv` doesn't seem like a good name for a hypothesis, first because it conflicts with the `triv` tactic, and second because these results are clearly nontrivial. Names for hypotheses starting with `h` and at most two or three characters long are preferred.